core/src/main/java/hudson/node_monitors/NodesDetailsUsageStatisticsContributor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2016, Kohsuke Kawaguchi, Baptiste Mathus
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.node_monitors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hudson.Extension;
+import hudson.model.Computer;
+import hudson.model.UsageStatisticsContributor;
+import hudson.node_monitors.ArchitectureMonitor.DescriptorImpl;
+import jenkins.model.Jenkins;
+import net.sf.json.JSONObject;
+
+/**
+ * Publishes informations about the used VM, number of executors, OS and so on
+ * for each node of the Jenkins cluster to the usage statistics subsystem.
+ *
+ * @see ArchitectureMonitor
+ */
+@Extension
+public class NodesDetailsUsageStatisticsContributor extends UsageStatisticsContributor {
+
+    @Override
+    public JSONObject getStatData() {
+        Jenkins j = Jenkins.getActiveInstance();
+        JSONObject o = new JSONObject();
+        List<JSONObject> nodes = new ArrayList<JSONObject>();
+        for (Computer c : j.getComputers()) {
+            JSONObject n = new JSONObject();
+            if (c.getNode() == j) {
+                n.put("master", true);
+                n.put("jvm-vendor", System.getProperty("java.vm.vendor"));
+                n.put("jvm-name", System.getProperty("java.vm.name"));
+                n.put("jvm-version", System.getProperty("java.version"));
+            }
+            n.put("executors", c.getNumExecutors());
+            DescriptorImpl descriptor = j.getDescriptorByType(DescriptorImpl.class);
+            n.put("os", descriptor.get(c));
+            nodes.add(n);
+        }
+        o.put("nodes", nodes);
```

Comment:
Right. IIUC, you mean that basically only L59&L60 should really be _there_, and the rest be left in the `CoreUsageStatistics` class?

Since only those lines
```
DescriptorImpl descriptor = j.getDescriptorByType(DescriptorImpl.class);
n.put("os", descriptor.get(c));
```
are indeed coupling UsageStatistics to `node_monitors`. 

Totally makes sense, I'll see how I can rework that in the next few days. Cause indeed this wouldn't be an issue in short term as we plan to bundle the extracted plugin anyway, but would be a bigger risk to lower the richness of retrieved stats in the future once that plugin may not be bundled anymore.